### PR TITLE
fix filter index for ov::element::u1 in BinaryConvolution

### PR DESCRIPTION
**Pull request by wilson-seok**
### Details:
 - fix filter index for ov::element::u1 in BinaryConvolution
 - add input range for BinaryConvolution in share_test_classes

### Tickets:
 - 82034

### CODE REVIEW

**Review by iefode — APPROVED** [review]
LGTM from IE test infra side

**Review by iefode — APPROVED** [review]
LGTM from Ie test infra perspective

**Comment by iefode** [comment]
@ilyachur Could you please take a look to reference changes?

File: src/core/reference/include/ngraph/runtime/reference/binary_convolution.hpp [hunk 1]
```diff
@@ -130,7 +130,7 @@ void binary_convolution(const T_IN* in,
 
     const size_t filters_count = filters_shape[filter_out_ch_axis];
     const Shape filter_shape(++filters_shape.begin(), filters_shape.end());
-    const size_t filter_size = shape_size(filter_shape);
+    const size_t filter_size = shape_size(filter_shape) / 8;
```
Comment (ilyachur) [inline src/core/reference/include/ngraph/runtime/reference/binary_convolution.hpp:133]: What if filter_size % 8 != 0?

Comment (iefode) [reply src/core/reference/include/ngraph/runtime/reference/binary_convolution.hpp:133]: @wilson-seok

Comment (wilson-seok) [reply src/core/reference/include/ngraph/runtime/reference/binary_convolution.hpp:133]: @ilyachur I couldn't understand why filter_size %8 != 0 case is considered. I think filter_size doesn't matter %8 =0 or not. Is there any restrict?

Comment (ilyachur) [reply src/core/reference/include/ngraph/runtime/reference/binary_convolution.hpp:133]: @wilson-seok I would like to understand what if `shape_size(filter_shape) % 8 != 0`? Will it work? In this case We will round off the result

Comment (wilson-seok) [reply src/core/reference/include/ngraph/runtime/reference/binary_convolution.hpp]: @ilyachur I added the case of shape_size(filter_shape) % 8 != 0. Thanks!

**Review by ilyachur — APPROVED** [review]

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Pull request closed without merging**